src/main/scala/com/fulcrumgenomics/FgBioDef.scala

```diff
@@ -75,7 +75,7 @@ class FgBioDef extends CommonsDef {
     * @param f a function called on all values
     * @tparam T the type of the index
     */
-  @inline @specialized def forloop[T](from: T)(check: T => Boolean)(next: T => T)(f: T => Unit): Unit = {
+  @inline def forloop[@specialized T](from: T)(check: T => Boolean)(next: T => T)(f: T => Unit): Unit = {
```

Comment:
Only if you want it specialized for a sub-set of primitive types.  As is it will cause it to be specialized for all primitives.